src/domain/accounts/api-keys.ts

```diff
@@ -0,0 +1,55 @@
+import util from "util"
+import crypto, { BinaryLike } from "crypto"
+
+import {
+  ApiKeyError,
+  ApiKeyHashError,
+  InvalidApiKeyError,
+  InvalidExpirationError,
+} from "./errors"
+
+const randomBytes = util.promisify(crypto.randomBytes)
+const scrypt = util.promisify<BinaryLike, BinaryLike, number, Buffer>(crypto.scrypt)
+
+const randomString = async (size: number, encoding: BufferEncoding): Promise<string> => {
+  const rnd = await randomBytes(size)
+  return rnd.toString(encoding)
+}
+
+const hasExpired = (expireAt: Date): boolean => {
+  const now = new Date(Date.now())
+  return expireAt < now
+}
+
+export const randomApiKey = async (
+  expireAt: Date,
+  label?: string,
```

Comment:
Shouldn't be optional - defaults should be set higher up.

Reply:
fixed